mesonbuild/dependencies/dev.py

```diff
@@ -530,6 +547,38 @@ def __init__(self, environment: 'Environment', kwargs: T.Dict[str, T.Any]):
         java_home_include = self.java_home / 'include'
         self.compile_args.append(f'-I{java_home_include}')
         self.compile_args.append(f'-I{java_home_include / platform_include_dir}')
+
+        if modules:
+            java_home_lib = self.java_home / 'lib'
+
+            # Location of libraries is different in JDK < 9 due to the JRE.
+            if version_compare(self.version, '<= 1.8.0'):
+                # The JDK and Meson have a disagreement here, so translate it
+                # over. In the event more translation needs to be done, add to
+                # following dict.
+                def cpu_translate(cpu: str) -> str:
+                    java_cpus = {
+                        'x86_64': 'amd64'
+                    }
+
+                    return java_cpus.get(cpu, cpu)
+
+                translated_cpu = cpu_translate(m.cpu_family)
+
+                if 'jvm' in modules:
+                    java_home_jre_lib_cpu_server = self.java_home / "jre" / "lib" / translated_cpu / 'server'
```

Comment:
In the previous version, `'server'` was only part of the JDK 9 `else` clause. Was this an intentional fix compared to the previous patch revision?

Reply:
Yes. This fixed the CI failure which I also noticed on my local machine to confirm after switching javac to 1.8.0

Reply:
If you have a Java 8 on your system, it is easy to confirm. I keep `latest`, `11`, and `1.8` installed on my system.

`find /usr/lib/jvm/` -type f -name "libjvm.so"`. Switch `jvm` from `jawt` to view the other library.

Reply:
I have 11 as my default, so I didn't notice the 1.8 failure initially although I obviously planned for it :)

Reply:
Ah, neat. So what this means then is that we can simplify a lot of this. All we need to do is define `java_home_lib` depending on the java version, e.g. for < 1.8.0 we define `java_home_lib = self.java_home / "jre" / "lib" / cpu_translate(m.cpu_family)`.

Then there is no version difference for the rest, so we leave the version checking if/else and, for all versions of java, extend the link args with:

```python
if 'jvm' in modules:
    self.link_args.extend([f'-L{java_home_lib / "server"}', '-ljvm'])
if 'awt' in modules:
    self.link_args.extend([f'-L{java_home_lib}', '-ljawt'])
```

Reply:
Good point. Let me go fix this up.